custom_components/tahoma/cover.py

```diff
@@ -211,39 +209,26 @@ def set_cover_tilt_position(self, **kwargs):
     def is_closed(self):
         """Return if the cover is closed."""
 
-        if "core:OpenClosedState" in self.tahoma_device.active_states:
-            return (
-                self.tahoma_device.active_states.get("core:OpenClosedState") == "closed"
-            )
-
-        if "core:SlatsOpenClosedState" in self.tahoma_device.active_states:
-            return (
-                self.tahoma_device.active_states.get("core:SlatsOpenClosedState")
-                == "closed"
-            )
-
-        if "core:OpenClosedPartialState" in self.tahoma_device.active_states:
-            return (
-                self.tahoma_device.active_states.get("core:OpenClosedPartialState")
-                == "closed"
-            )
-
-        if "core:OpenClosedPedestrianState" in self.tahoma_device.active_states:
-            return (
-                self.tahoma_device.active_states.get("core:OpenClosedPedestrianState")
-                == "closed"
-            )
-
-        if "core:OpenClosedUnknownState" in self.tahoma_device.active_states:
-            return (
-                self.tahoma_device.active_states.get("core:OpenClosedUnknownState")
-                == "closed"
-            )
-
-        if getattr(self, "_position", None) is not None:
+        states = self.tahoma_device.active_states
+        if "core:OpenClosedState" in states:
+            return states.get("core:OpenClosedState") == "closed"
+
+        if "core:SlatsOpenClosedState" in states:
+            return states.get("core:SlatsOpenClosedState") == "closed"
+
+        if "core:OpenClosedPartialState" in states:
+            return states.get("core:OpenClosedPartialState") == "closed"
+
+        if "core:OpenClosedPedestrianState" in states:
+            return states.get("core:OpenClosedPedestrianState") == "closed"
+
+        if "core:OpenClosedUnknownState" in states:
+            return states.get("core:OpenClosedUnknownState") == "closed"
+
+        if self._position is not None:
             return self._position == 0
 
-        if getattr(self, "_tilt_position", None) is not None:
+        if self._tilt_position is not None:
```

Comment:
To be checked, but IMO the is_closed is only related to the position not the tilt position.

Reply:
I own a device (BioclimaticPergola) which only has slats that could be tilted (or closed). In that case only `tilt_position` exists.

Reply:
OK I understand. In home assistant, it will ease its usage indeed. In the UI, there is no button to close or open the tilt? And their is no visual indication if your pergola is opened or closed without this trick?

Reply:
There is a button to open and close the tilt, see `open_cover_tilt` and `close_cover_tilt`. Based on SUPPORTED_FEATURES the required buttons are shown in HA.

Reply:
Toggling SUPPORTED_FEATURES based on the available commands, this is very dynamic and supports multiple device types without any hardcoded list.

Reply:
How does it look like without this fix? Sorry for my question, but often due to little hack, it's hard to understand the code later.

Reply:
It is not really a hack, it first does the check if it is closed based on the state. If the state is not available, it has a look at the current position. If the current position is unknown, it falls back to the tilt position.